src/connection/server_support.c

```diff
@@ -2330,21 +2330,21 @@ css_change_ha_server_state (THREAD_ENTRY * thread_p, HA_SERVER_STATE state, bool
 	  break;
 	}
       /* If log appliers have changed their state to done, go directly to active mode */
-      if (css_check_ha_log_applier_done ())
+      if (!HA_DISABLED ())
```

Comment:
The 'force' branch of code may seem duplicate.
What about relocating non-duplicate code of 'force' after the switch and let the 'switch' handle the common part ?